StitchCore/Common/Builder.swift

```diff
@@ -0,0 +1,34 @@
+/**
```

Comment:
[q] How useful is this abstraction?

Reply:
This was a nested abstraction I had in core previously– Adam has simply moved it out to its own file.

With the builder pattern we are using, it saves us some repeated code for each builder, which we have a fair amount of. It's also the only way to create conformance with this kind of abstraction.

This was especially helpful with all of the builder extensions we wrote, based off of the Android builders.

The alternative is to do a more "old school" (less Swifty, more Java-y) builder pattern and do a fair amount of copy and pasting.

Reply:
Going to leave as is then